sn_node/src/put_validation.rs

```diff
@@ -385,29 +389,53 @@ impl Node {
         Ok(CmdOk::StoredSuccessfully)
     }
 
-    /// Gets CashNotes out of a Payment, this includes network verifications of the Transfer
+    /// Gets CashNotes out of a Payment, this includes network verifications of the Transfer.
+    /// Return the CashNotes corresponding to our wallet and the ones corresponding to network royalties payment.
     async fn cash_notes_from_payment(
         &self,
         payment: &Vec<Transfer>,
         wallet: &LocalWallet,
         pretty_key: PrettyPrintRecordKey,
-    ) -> Result<(Vec<CashNote>, Transfer), ProtocolError> {
+    ) -> Result<(Vec<CashNote>, Vec<CashNote>), ProtocolError> {
+        let royalties_pk = *NETWORK_ROYALTIES_PK;
+        let mut cash_notes = vec![];
+        let mut royalties_cash_notes = vec![];
+
         for transfer in payment {
-            match self
-                .network
-                .verify_and_unpack_transfer(transfer, wallet)
-                .await
-            {
-                // transfer not for us
-                Err(ProtocolError::FailedToDecypherTransfer) => continue,
-                // transfer invalid
-                Err(e) => return Err(e),
-                // transfer ok
-                Ok(cash_notes) => return Ok((cash_notes, transfer.clone())),
-            };
+            match transfer {
+                Transfer::Encrypted(_) => match self
+                    .network
+                    .verify_and_unpack_transfer(transfer, wallet)
+                    .await
+                {
+                    // transfer not for us
+                    Err(ProtocolError::FailedToDecypherTransfer) => continue,
+                    // transfer invalid
+                    Err(e) => return Err(e),
+                    // transfer ok
+                    Ok(cn) => cash_notes = cn,
```

Comment:
this will always only take the cash_notes list from the last payment.
is it ok ?

Reply:
I assume so, it used to take just the first one before this change, from my understanding there will be only one that will get through though which is the one corresponding to the node trying to decrypt the transfer.